File: fibonacciJava/java/src/com/company/Fibonacci.java
package com.company;

public class Fibonacci {
    public static int valueOfIndex(int index) {
        int value = 0;

        if (index < 0){
            throw new IllegalArgumentException("index must be non-negative.");
        }

        if (index == 0 || index == 1){
            value = index;
        }

        if (index > 1 && index <= 46){
           int preOne = 1;
           int preTwo = 0;
           for (int i = 2; i <= index; i++){
               value = preOne + preTwo;
               preTwo = preOne;
               preOne = value;
           }
        }

        if (index >= 47){
            throw new IllegalArgumentException("index is too big.");
        }

        return value;
    }
}

File: fibonacciJavascript/fibonacci.js
var input = prompt("input an index: ");
if (validate(input)){
    var value = fibonacci(input * 1);
    if (value === Infinity){
        alert("index is too big.");
    }
    else{
        alert("the value of index " + input + " is " + value);
    }
}

function validate(message){
        if (isNaN(message) || message < 0){
            alert("index must be a non-negative integer.");
            return false;
        }
        for (c of message){
            if (c == "."){
                alert("index must be an integer.");
            return false;
            }
        }
    return true;
}

function fibonacci(index){
    if (index <= 1){
        return index;
    }

    var fibonacciValue;
    var preOne = 1;
    var preTwo = 0;
    for (let i = 2; i <= index; i++){
        fibonacciValue = preOne + preTwo;
        preTwo = preOne;
        preOne = fibonacciValue;
    }
    return fibonacciValue;
}